### PR TITLE
use go-padreader instead of local copy

### PR DESCRIPTION
**Pull request by laser**
I've extracted go-padreader into a new repo so that it can be shared by other projects (lotus, go-storage-miner, and this project).

This changeset backports that extraction into go-fil-markets.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -680,6 +684,8 @@ gopkg.in/yaml.v2 v2.2.2 h1:ZCJp+EgiOT7lHqUV2J862kp8Qj64Jo6az82+3Td9dZw=
 gopkg.in/yaml.v2 v2.2.2/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 gopkg.in/yaml.v2 v2.2.4 h1:/eiJrUcujPVeJ3xlSWaiNi3uSVmDGBK1pDHUHAnao1I=
 gopkg.in/yaml.v2 v2.2.4/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
+gotest.tools v2.2.0+incompatible h1:VsBPFP1AI068pPrMxtb/S8Zkgf9xEmTLJjfM+P5UIEo=
+gotest.tools v2.2.0+incompatible/go.mod h1:DsYFclhRJ6vuDpmuTbkuFWG+y2sxOXAzmJt81HFBacw=
```
Comment (ingar) [inline go.sum]: Also, don't know if you intended to use `gotools.test` in the extracted padreader.

Comment (laser) [reply go.sum]: @ingar - nope, didn't: https://github.com/filecoin-project/go-padreader/pull/1

Comment (laser) [reply go.sum]: updated and merged

**Comment by laser** [comment]
update: I've deleted the generated PadReader.go mock

**Comment by laser** [comment]
thnx

**Pull request merged**